From 421655c282bc6dcf4a79d148fe46ab3b83aa6808 Mon Sep 17 00:00:00 2001
From: debbuild <debbuild>
Date: Sat, 20 Oct 2018 10:56:03 +0200
Subject: [PATCH] Update CWEBbin manpage.

Add new options for cweave in the CWEBbin extension.
Use some formatting ideas from cweb.1 in TeXlive.
STAT got lost in CWEB 3.0.
---
 cweb.1 | 41 ++++++++++++++++++++++++-----------------
 1 file changed, 24 insertions(+), 17 deletions(-)

diff --git a/cweb.1 b/cweb.1
index ad41c42..80c9dd4 100644
--- a/cweb.1
+++ b/cweb.1
@@ -1,4 +1,4 @@
-.TH CWEB 1L 2002-Apr-13
+.TH CWEB 1 "20 October 2018" "CWEBbin 2018"
 .
 .SH NAME
 ctangle, cweave \- translate CWEB to C and/or TeX
@@ -10,14 +10,16 @@ ctangle, cweave \- translate CWEB to C and/or TeX
 .B \-bhp
 ] [
 .B +s
-] webfile[.w] [changefile[.ch]] [outputfile[.c]]
+] webfile[.w] [{changefile[.ch]|-} [outputfile[.c]]]
 .br
 .B cweave
 [
-.B \-befhpx
+.B \-bfhiopx
 ] [
-.B +s
-] webfile[.w] [changefile[.ch]] [outputfile[.tex]]
+.B \-lX
+] [
+.B +es
+] webfile[.w] [{changefile[.ch]|-} [outputfile[.tex]]]
 .ad
 .
 .SH DESCRIPTION
@@ -78,10 +80,9 @@ If you say -bhp, you get nothing but error messages.
 .PP
 The
 .B +s
-option prints statistics about memory usage at the end of a run
-(assuming that the programs have been compiled with the -DSTAT switch).
+option prints statistics about memory usage at the end of a run.
 .PP
-There are three other options applicable to
+There are six other options applicable to
 .I cweave
 only:
 .B \-f
@@ -95,24 +96,29 @@ Such brackets are normally inserted so that special hooks
 can be used by
 .I cweb-latex
 and similar programs.
+.B \-i
+suppresses the indentation of parameter declarations.
+.B \-lX
+uses macros for language X\ from Xcwebmac.tex.
+.B \-o
+suppresses the visual separation of declarations and statements.
 .B \-x
 means omit the index and table of contents.
 .
+.SH ENVIRONMENT
+The environment variable CWEBINPUTS is used to search for the input files,
+or the system default if CWEBINPUTS is not set.
+.
 .SH FILES
+The location of the files mentioned below varies from system to system.
 .TP
-/usr/local/lib/tex/inputs/cwebmac.tex
+cwebmac.tex
 TeX macros used by cweave output.
 .TP
-/usr/local/src/cweb/cwebman.tex
+cwebman.tex
 The user manual.
 .TP
-/usr/local/src/cweb/examples/wc.w
-An introductory example.
-.TP
-/usr/local/src/cweb/examples/wmerge.w
-Patch program based on CWEB-style change files.
-.TP
-/usr/local/lib/cweb
+/usr/lib/cweb
 Directory for cweb "include" files.
 .
 .SH "SEE ALSO"
@@ -135,4 +141,5 @@ Silvio Levy designed and developed CWEB
 by adapting the WEB conventions to\ C and by recoding everything in CWEB.
 Knuth began using CWEB and made further refinements.
 Many other helpers are acknowledged in the CWEB manual.
+CWEBbin was developed by Andreas Scherer.
 
-- 
2.19.1

